aten/src/ATen/native/UpSample.h

```diff
@@ -111,11 +111,22 @@ static inline void upsample_3d_shape_check(
   }
 }
 
+template <typename scalar_t>
+static inline scalar_t compute_scales_value(
+    const double scale,
+    int64_t input_size,
+    int64_t output_size) {
+      return (scale > 0.)
+          ? static_cast<scalar_t>(1. / static_cast<scalar_t>(scale))
+          : (static_cast<scalar_t>(input_size) / output_size);
```

Comment:
`input_size` and `output_size` are integers. Should we cast them as float first before division to get floating point division and output. That will be more accurate for scale.

Reply:
Actually the division is between static_cast<scalar_t>(input_size) and output_size. 
So input_size is converted to scalar_t (which would be float or double), then divided by output_size (int).
(The division is between a float/double and an int, so the compiler would convert output_size to scalar_t as well.)